src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java

```diff
@@ -60,11 +60,19 @@
      *
      * @author James E. Blair &lt;jeblair@hp.com&gt;
      */
-    private class GerritTriggeredEventComparator implements Comparator<GerritTriggeredEvent> {
+    static class GerritTriggeredEventComparator implements Comparator<GerritTriggeredEvent> {
         @Override
         public int compare(GerritTriggeredEvent o1, GerritTriggeredEvent o2) {
-            return new Integer(((java.lang.Object)o1).hashCode()).compareTo(
-                    new Integer(((java.lang.Object)o2).hashCode()));
+            if (o1 == null && o2 == null) {
+                return 0;
+            }
+            if (o1 != null && o2 == null) {
+                return 1;
+            }
+            if (o1 == null && o2 != null) {
+                return -1;
+            }
+            return new Integer(o1.hashCode()).compareTo(o2.hashCode());
```

Comment:
I know this is an aside from the current issue, but comparing events based on a 32 bit integer in order to 

> This ensures that every event received from Gerrit is kept track of individually

Seems a bit naïve